pyproject.toml

```
@@ -1,21 +1,150 @@
+# However, you should not include napari[all], napari[qt],
+# or any other Qt bindings directly (e.g. PyQt5, PySide2).
+# See best practices: https://napari.org/stable/plugins/building_a_plugin/best_practices.html
+dependencies = [
+    "numpy",
+    "magicgui",
+    "napari",
+    "qtpy",
+    "scikit-image",
+    "tiled[array,minimal-client]",
+]
+
+[project.optional-dependencies]
+# Allow easily installation with the full, default napari installation
+# (including Qt backend) using napari-tiled[all].
+all = ["napari[all]"]
+dev = [
+    "napari[qt]",
+    "PyQt5"
+]
+testing = [
+    "tox",
+    "pytest",  # https://docs.pytest.org/en/latest/contents.html
+    "pytest-cov",  # https://pytest-cov.readthedocs.io/en/latest/
+    "pytest-qt",  # https://pytest-qt.readthedocs.io/en/latest/
+    "napari[qt]",  # test with napari's default Qt bindings
+]
+
+[project.entry-points."napari.manifest"]
+napari-tiled = "napari_tiled_browser:napari.yaml"
+
+[project.urls]
+"Bug Tracker" = "https://github.com/AbbyGi/napari-tiled/issues"
+"Documentation" = "https://github.com/AbbyGi/napari-tiled#README.md"
+"Source Code" = "https://github.com/AbbyGi/napari-tiled"
+"User Support" = "https://github.com/AbbyGi/napari-tiled/issues"
+
 [build-system]
-requires = ["setuptools>=42.0.0", "wheel"]
+requires = ["setuptools>=42.0.0", "setuptools_scm"]
 build-backend = "setuptools.build_meta"
 
-[project]
-name = "napari-tiled"
-version = "0"  # HACK
-dependencies = ["tiled[array,minimal-client]"]
+[tool.setuptools]
+include-package-data = true
+
+[tool.setuptools.packages.find]
+where = ["src"]
 
-[options.entry_points]
-napari.manifest = "napari_tiled_browser:napari.yaml"
+[tool.setuptools.package-data]
+"*" = ["*.yaml"]
 
-[options.package_data]
-napari-tiled = "napari.yaml"
+[tool.setuptools_scm]
+write_to = "src/napari_tiled_browser/_version.py"
+fallback_version = "0.0.1+nogit"
 
 [tool.black]
 line-length = 79
+target-version = ['py310', 'py311', 'py312', 'py313']
+
+[tool.ruff]
+line-length = 79
+lint.select = [
+    "E", "F", "W", #flake8
+    "UP", # pyupgrade
+    "I", # isort
+    "BLE", # flake8-blind-exception
+    "B", # flake8-bugbear
+    "A", # flake8-builtins
+    "C4", # flake8-comprehensions
+    "ISC", # flake8-implicit-str-concat
+    "G", # flake8-logging-format
+    "PIE", # flake8-pie
+    "SIM", # flake8-simplify
+]
+lint.ignore = [
+    "E501", # line too long. let black handle this
+    "UP006", "UP007", # type annotation. As using magicgui require runtime type annotation then we disable this.
+]
+
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".mypy_cache",
+    ".pants.d",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "venv",
+    "*vendored*",
+    "*_vendor*",
+]
+
+target-version = "py310"
+fix = true
+
+[tool.pixi.workspace]
+channels = ["conda-forge"]
+platforms = ["osx-64"]
```

Adding linux platform will let it run on VDI.  Might as well add the other major platforms.

```suggestion
platforms = ["linux-64", "osx-64", "osx-arm64", "win-64"]
```